src/internal/packager2/layout/package.go

```diff
@@ -0,0 +1,306 @@
+		dir := filepath.Dir(header.Name)
+		if dir != "." {
+			err := os.MkdirAll(filepath.Join(dirPath, dir), helpers.ReadExecuteAllWriteUser)
+			if err != nil {
+				return err
+			}
+		}
+		dst, err := os.Create(filepath.Join(dirPath, header.Name))
+		if err != nil {
+			return err
+		}
+		defer dst.Close()
+		_, err = io.Copy(dst, f)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	p, err := LoadFromDir(ctx, dirPath, opt)
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+// LoadFromDir loads and validates a package from the given directory path.
+func LoadFromDir(ctx context.Context, dirPath string, opt PackageLayoutOptions) (*PackageLayout, error) {
+	b, err := os.ReadFile(filepath.Join(dirPath, ZarfYAML))
+	if err != nil {
+		return nil, err
+	}
+	var pkg v1alpha1.ZarfPackage
+	err = goyaml.Unmarshal(b, &pkg)
+	if err != nil {
+		return nil, err
+	}
+	if len(pkg.Build.Migrations) > 0 {
+		for idx, component := range pkg.Components {
+			pkg.Components[idx], _ = deprecated.MigrateComponent(pkg.Build, component)
+		}
+	}
+
+	pkgLayout := &PackageLayout{
+		dirPath: dirPath,
+		Pkg:     pkg,
+	}
+	err = validatePackageIntegrity(pkgLayout, opt.IsPartial)
+	if err != nil {
+		return nil, err
+	}
+	err = validatePackageSignature(ctx, pkgLayout, opt.PublicKeyPath, opt.SkipSignatureValidation)
+	if err != nil {
+		return nil, err
+	}
+	return pkgLayout, nil
+}
+
+// Cleanup removes any temporary directories created.
+func (p *PackageLayout) Cleanup() error {
+	err := os.RemoveAll(p.dirPath)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetSBOM outputs the SBOM data from the package to the give destination path.
+func (p *PackageLayout) GetSBOM(destPath string) (string, error) {
+	path := filepath.Join(destPath, p.Pkg.Metadata.Name)
+	err := archiver.Extract(filepath.Join(p.dirPath, SBOMTar), "", path)
+	if err != nil {
+		return "", err
+	}
+	return path, nil
+}
+
+// GetComponentDir returns a path to the directory in the given component.
+func (p *PackageLayout) GetComponentDir(destPath, componentName string, ct ComponentDir) (string, error) {
+	sourcePath := filepath.Join(p.dirPath, ComponentsDir, fmt.Sprintf("%s.tar", componentName))
+	_, err := os.Stat(sourcePath)
+	if err != nil {
+		return "", fmt.Errorf("component %s does not exist in package: %w", componentName, err)
+	}
+	tmpDir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return "", err
+	}
+	defer os.RemoveAll(tmpDir)
+	// TODO (phillebaba): We are not using archiver.Extract here because there is a bug in Windows where the files will not be extracted properly from nested directories.
+	err = archiver.Unarchive(sourcePath, tmpDir)
+	if err != nil {
+		return "", err
+	}
+	compPath := filepath.Join(tmpDir, componentName, string(ct))
+	_, err = os.Stat(compPath)
+	if err != nil {
+		return "", fmt.Errorf("component %s does not have a %s directory: %w", componentName, ct, err)
```

Comment:
```suggestion
		return "", fmt.Errorf("component %s could not access %s directory: %w", componentName, ct, err)
```
os.Stat could return isnotfound or permissions error, want add clarity here